File: examples/docs-code-snippets/src/main/java/examples/java/application/FireSensoryEventAndAwaitReceived.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;
import com.ing.baker.runtime.javadsl.EventInstance;
import examples.java.events.OrderPlaced;

public class FireSensoryEventAndAwaitReceived {

    private final Baker baker;

    public FireSensoryEventAndAwaitReceived(Baker baker) {
        this.baker = baker;
    }

    public void example(String recipeInstanceId, OrderPlaced orderPlaced) {
        var eventInstance = EventInstance.from(orderPlaced);
        var sensoryEventStatus = baker.fireSensoryEventAndAwaitReceived(recipeInstanceId, eventInstance).join();
    }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/application/FireSensoryEventAndAwaitReceived.kt
package examples.kotlin.application

import com.ing.baker.runtime.javadsl.EventInstance
import com.ing.baker.runtime.kotlindsl.Baker
import examples.kotlin.events.OrderPlaced

class FireSensoryEventAndAwaitReceived(private val baker: Baker) {

    suspend fun example(recipeInstanceId: String, orderPlaced: OrderPlaced) {
        val orderPlacedEvent = EventInstance.from(orderPlaced)
        val sensoryEventStatus = baker.fireSensoryEventAndAwaitReceived(recipeInstanceId, orderPlacedEvent)
    }
}